src/avdecc/mcDomainManager.hpp

```diff
@@ -0,0 +1,121 @@
+/*
+* Copyright 2017-2018, Emilien Vallot, Christophe Calmejane and other contributors
+
+* This file is part of Hive.
+
+* Hive is free software: you can redistribute it and/or modify
+* it under the terms of the GNU Lesser General Public License as published by
+* the Free Software Foundation, either version 3 of the License, or
+* (at your option) any later version.
+
+* Hive is distributed in the hope that it will be usefu_state,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+* GNU Lesser General Public License for more details.
+
+* You should have received a copy of the GNU Lesser General Public License
+* along with Hive.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#pragma once
+
+#include <la/avdecc/controller/avdeccController.hpp>
+#include <memory>
+#include <optional>
+#include <QObject>
+#include <QMap>
+
+
+namespace avdecc
+{
+namespace mediaClock
+{
+using DomainIndex = std::uint64_t;
+
+
+// **************************************************************
+// class MCDomain
+// **************************************************************
+/**
+* @brief    Data container for domain data.
+* [@author  Marius Erlen]
+* [@date    2018-10-14]
+*/
+class MCDomain
+{
+public:
+	DomainIndex domainIndex() const;
+	void setDomainIndex(DomainIndex domainIndex);
+	QString displayName() const;
+	la::avdecc::UniqueIdentifier mediaClockDomainMaster() const;
+	void setMediaClockDomainMaster(la::avdecc::UniqueIdentifier entityId);
+	la::avdecc::entity::model::SamplingRate domainSamplingRate() const;
+	void setDomainSamplingRate(la::avdecc::entity::model::SamplingRate samplingRate);
+
+protected:
+	int _domainIndex;
+	la::avdecc::UniqueIdentifier _mcMaster;
+	la::avdecc::entity::model::SamplingRate _domainSamplingRate;
+};
+
+// **************************************************************
+// class MCEntityDomainMapping
+// **************************************************************
+/**
+* @brief    Data container for mc domains and assigned entities.
+* [@author  Marius Erlen]
+* [@date    2018-10-14]
+*/
+class MCEntityDomainMapping
+{
+public:
+	std::optional<DomainIndex> const findDomainIndexByMasterEntityId(la::avdecc::UniqueIdentifier mediaClockMasterId) noexcept;
+
+	std::map<la::avdecc::UniqueIdentifier, std::vector<DomainIndex>>& entityMediaClockMasterMappings();
+	std::map<DomainIndex, MCDomain>& mediaClockDomains();
+
+protected:
+	std::map<la::avdecc::UniqueIdentifier, std::vector<DomainIndex>> _entityMediaClockMasterMappings;
```

Comment:
Any reason why not using an unordered_map?

Reply:
This was improved in c6c49a8